serializers/src/main/java/io/pravega/schemaregistry/schemas/JSONSchema.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.schemas;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.module.jsonSchema.JsonSchema;
+import com.fasterxml.jackson.module.jsonSchema.JsonSchemaGenerator;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.contract.data.SerializationFormat;
+import lombok.Getter;
+import org.apache.avro.specific.SpecificRecordBase;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Container class for Json Schema.
+ *
+ * @param <T> Type of element. 
+ */
+public class JSONSchema<T> implements Schema<T> {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Getter
+    private final String schemaString;
+    private final Class<T> base;
+    @Getter
+    private final Class<? extends T> derived;
+    
+    @Getter
+    private final JsonSchema schema;
+
+    private final SchemaInfo schemaInfo;
+
+    private JSONSchema(JsonSchema schema, String name, String schemaString, Class<T> derived) {
+        this(schema, name, schemaString, derived, derived);
+    }
+
+    private JSONSchema(JsonSchema schema, String name, String schemaString, Class<T> base, Class<? extends T> derived) {
+        this.schemaString = schemaString;
+        this.schemaInfo = new SchemaInfo(name, SerializationFormat.Json, getSchemaBytes(), ImmutableMap.of());
+        this.base = base;
+        this.derived = derived;
+        this.schema = schema;
+    }
+
+    private JSONSchema(SchemaInfo schemaInfo, JsonSchema schema, String schemaString, Class<T> derived) {
+        this.schemaString = schemaString;
+        this.schemaInfo = schemaInfo;
+        this.base = derived;
+        this.derived = derived;
+        this.schema = schema;
+    }
+
+    /**
+     * Method to create a typed JSONSchema for the given class. It extracts the json schema from the class.
+     * For POJOs the schema is extracted using jacksons {@link JsonSchemaGenerator}. 
```

Comment:
`jacksons` -> `jackson`. maybe add a link reference too.

Reply:
changing it to apostrophe `jackson's`